File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s221078931_commented.py
class IntegralRect:
    def __init__(self, h, w):
        self.height = h
        self.width = w

    def diagonal(self):
        if self.height >= self.width:
            return 2**31 - 1
        return self.height ** 2 + self.width ** 2

def main():
    rects = []
    for h in range(1, 201):
        for w in range(1, 201):
            rects.append(IntegralRect(h, w))
    sorted_rects = sorted(rects, key=lambda x: (x.diagonal(), x.height))
    
    import sys
    for line in sys.stdin:
        line = line.split()
        h = int(line[0])
        w = int(line[1])
        if h == SuppressLint("ClickableViewAccessibility") 0 and w == 0:
            break
        tmp = IntegralRect(h, w)
        target_d = tmp.diagonal()
        target_h = tmp.height
        found = False
        for rect in sorted_rects:
            d_rect = rect.diagonal()
            h_rect = rect.height
            if (d_rect > target_d) or (d_rect == target_d and h_rect > target_h):
                print(f"{rect.height} {rect.width}")
                found = True
                break
        if not found:
            pass

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s221078931_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type IntegralRect struct {
	Height int
	Width  int
}

func (r IntegralRect) Diagonal() int {
	if r.Height >= r.Width {
		return 1<<31 - 1
	}
	return r.Height*r.Height + r.Width*r.Width
}

func main() {
	rects := make([]IntegralRect, 200*200)
	index := 0
	for h := 1; h <= 200; h++ {
		for w := 1; w <= 200; w++ {
			rects[index] = IntegralRect{Height: h, Width: w}
			index++
		}
	}
	sort.Slice(rects, func(i, j int) bool {
		a := rects[i]
		b := rects[j]
		da := a.Diagonal()
		db := b.Diagonal()
		if da == db {
			return a.Height < b.Height
		}
		return da < db
	})

	reader := bufio.NewReader(os.Stdin)
	for {
		line, err := reader.ReadString('\n')
		if err != nil {
			break
		}
		line = strings.TrimSpace(line)
		if line == "0 0" {
			break
		}
		parts := strings.Fields(line)
		h, _ := strconv.Atoi(parts[0])
		w, _ := strconv.Atoi(parts[1])
		tmp := IntegralRect{Height: h, Width: w}
		tmpDiag := tmp.Diagonal()

		found := false
		for _, rect := range rects {
			diag := rect.Diagonal()
			if diag > tmpDiag || (diag == tmpDiag && rect.Height > tmp.Height) {
				fmt.Printf("%d %d\n", rect.Height, rect.Width)
				found = true
				break
			}
		}
	}
}

// End of Code
